plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/preferences/OCBinary.java

```diff
@@ -139,4 +146,26 @@ public boolean isCompatibleForPublishing(IProgressMonitor monitor) {
 	public boolean isCompatibleForPublishing(IConnection connection, IProgressMonitor monitor) {
 	    return new OCBinaryVersionValidator(getLocation(connection)).isCompatibleForPublishing(monitor);
 	}
+	
+	/**
+	 * Compute the error message for the OCBinary state and path.
+	 * 
+	 * @param valid if the oc binary is valid or not
+	 * @param location the location of the oc binary
+	 * @return the error message (may be null)
+	 */
+	public IStatus getOCBinaryStatus(IConnection connection, IProgressMonitor monitor) {
+		String location = getLocation(connection);
+	    IStatus status = Status.OK_STATUS;
+	    if (!isCompatibleForPublishing(connection, monitor)) {
```

Comment:
why check the version 1st and the location afterwards? To me inversing this sequence makes more sense (1. Is a path provided? 2. does the path exist? 3. Does the version match the requirements)?

Reply:
makes sense, but it was just copied from https://github.com/jbosstools/jbosstools-openshift/pull/1550/files/4343beba8643b34ba04108828c17db2ebcf0ca5f#diff-5627fe7968b85348bfd960f2e0beae9aL206